From 0e165d630d8bf9fb35b1204dec99a97adcb5018a Mon Sep 17 00:00:00 2001
Message-Id: <0e165d630d8bf9fb35b1204dec99a97adcb5018a.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Mon, 7 Jul 2014 15:49:08 +0100
Subject: [PATCH 142/312] FOR_UPSTREAM [VPG]: drm/i915: Add
 WaDisableLSQCROPERFforOCL:bdw

L3SQCREG4 LQSC RO PERF DIS must be programmed by software to 1h
(Disable) to work around a Gsync Issue in HDC.

Issue: VIZ-4086
Change-Id: Id0fbcee8d70e04bcd4fd3468d96e118b4716dd95
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    1 +
 drivers/gpu/drm/i915/intel_pm.c |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 7b72b08..930c2c7 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4754,6 +4754,7 @@ enum punit_power_well {
 #define  GEN8_TAG_CLK_OFFTIME_MASK		(~((1<<23) | (1<<22) | (1<<21) | (1<<20)))
 
 #define GEN8_L3SQCREG4				0xb118
+#define  GEN8_L3SQCREG4_LQSC_RO_PERF_DISABLE	(1<<27)
 #define  GEN8_PIPELINE_FLUSH_COHERENT_LINES	(1<<21)
 
 /* WaCatErrorRejectionIssue */
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 912b9ed..da67486 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -5617,8 +5617,10 @@ static void gen8_init_clock_gating(struct drm_device *dev)
 	I915_WRITE(GEN7_MISCCPCTL, I915_READ(GEN7_MISCCPCTL) &
 			~GEN8_DOP_CLOCK_GATE_CFCLK_ENABLE);
 
+	/* WaDisableLSQCROPERFforOCL:bdw */
 	/* WaFlushCoherentL3CacheLinesAtContextSwitch:bdw */
 	I915_WRITE(GEN8_L3SQCREG4, I915_READ(GEN8_L3SQCREG4) |
+			GEN8_L3SQCREG4_LQSC_RO_PERF_DISABLE |
 			GEN8_PIPELINE_FLUSH_COHERENT_LINES);
 
 	/*
-- 
1.7.9.5

